app/src/apdu_handler.c

```diff
@@ -138,13 +160,31 @@ __Z_INLINE void handleGetAddr(volatile uint32_t *flags, volatile uint32_t *tx, u
     zemu_log("handleGetAddr\n");
 
     const uint8_t requireConfirmation = G_io_apdu_buffer[OFFSET_P1];
-    zxerr_t zxerr = fill_address((uint32_t *)flags, (uint32_t*)tx, rx, G_io_apdu_buffer, IO_APDU_BUFFER_SIZE);
+    const uint8_t curve_type = G_io_apdu_buffer[OFFSET_P2];
+
```

Comment:
I think it's better to add a curve type checker here
```suggestion
if (curve_type != CURVE_SECP256K1 && curve_type != CURVE_ED25519) {
    THROW(APDU_CODE_CONDITIONS_NOT_SATISFIED);
}
```